src/App.js

```diff
@@ -1,62 +1,80 @@
+import { useCallback, useState } from "react";
+
 import "./App.css";
-import { useState } from "react";
 import TodoList from "./TodoList";
-import TodoInput from "./TodoInput";
 
-// setTodos((previousState) => ({ ...previousState, status: done }));
 
 function App() {
   const [todos, setTodos] = useState([
     {
       id: "1",
       text: "buy some stuff",
-      status: "done",
+      isDone: true,
     },
     {
       id: "2",
       text: "finish home work",
-      status: "todo",
+      isDone: false,
     },
   ]);
   const [newTodoText, setNewTodoText] = useState("");
-  const deleteTodo = (id) => {
-    setTodos(todos.filter((todo) => todo.id != id));
-  };
-  const onChangeStatus = (id) => {
-    let changedTodos = todos.filter((todo) =>
-      todo.id == id
-        ? (todo.status = todo.status == "done" ? "todo" : "done")
-        : todo
+  const handleDeleteTodo = useCallback(
+    (id) =>
+      setTodos((previousTodos) =>
+        previousTodos.filter((todo) => todo.id !== id)
+      ),
+    []
+  );
+  const handleChangeStatus = useCallback((id) => {
+    setTodos((previousTodos) =>
+      previousTodos.map((todo) =>
+        todo.id === id ? { ...todo, isDone: !todo.isDone } : todo
+      )
     );
-    setTodos(changedTodos);
-    console.log(todos);
-  };
-  const setNewInputText = (e) => {
-    setNewTodoText(e.target.value);
-  };
-  const AddNewTodo = (e) => {
-    e.preventDefault();
-    if (newTodoText) {
-      setTodos((previousState) => [
-        ...previousState,
-        {
-          id: Date.now() + "",
-          text: newTodoText,
-          status: "todo",
-        },
-      ]);
-      e.target[0].value = "";
-    }
-    setNewTodoText("");
-  };
+  }, []);
+  const handleNewTodoText = useCallback((e) => {
```

Comment:
```suggestion
  const handleNewTodoText = useCallback((event) => {
```
always avoid abbreviations